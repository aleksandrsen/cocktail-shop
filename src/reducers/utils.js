export const resourceStartRecord = () => ({
    loading: false,
    loaded: false,
    error: null,
    entities: {}
});

export const arrToObj = arr => {
    return arr.reduce((init, item) => {
        init[item.id] = item;
        return init;
    }, {});
};