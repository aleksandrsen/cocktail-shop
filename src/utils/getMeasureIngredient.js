export default function getMeasureIngredient(obj) {
    let measure = [];
    if (!obj.strMeasure1) return null;
    for (let key in obj) {
        if (key.slice(0, -1) === "strMeasure" && obj[key] !== null) {
            measure.push(obj[key]);
        }
    }
    return measure;
}