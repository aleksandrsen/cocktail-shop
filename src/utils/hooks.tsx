import { RefObject, useCallback, useEffect, useState } from "react";
import { isInViewport } from "./index";
import {useLocation} from "react-router-dom";

export const useMedia = (
  queries: string[],
  values: number[],
  defaultValue: number
) => {
  const mediaQueryLists = queries.map((q) => window.matchMedia(q));

  const getValue = useCallback(() => {
    const index = mediaQueryLists.findIndex((mql) => mql.matches);
    return values[index] ? values[index] : defaultValue;
  }, [mediaQueryLists, values, defaultValue]);

  const [value, setValue] = useState(getValue);

  useEffect(() => {
    const handler = () => setValue(getValue);
    mediaQueryLists.forEach((mql) => mql.addListener(handler));
    return () => mediaQueryLists.forEach((mql) => mql.removeListener(handler));
  }, [getValue, mediaQueryLists]);

  return value;
};

export const useIsShowAnimation = (ref: RefObject<any>, name?: string) => {
  const [isShow, setIsShow] = useState(true);

  const location = useLocation();

  useEffect(() => {
    if (ref?.current) {

      const windowOffset = window.pageYOffset;

      console.log(windowOffset, "windowOffset")
      console.log(name, isInViewport(ref.current), ref.current.offsetTop, windowOffset)

      if (isInViewport(ref.current) || ref.current.offsetTop <= windowOffset) {
        setIsShow(false);
        console.log("inside")
      }

    }
  }, [ref?.current, location]);

  return isShow;
};
