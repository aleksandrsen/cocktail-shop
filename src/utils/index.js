export const cutTextContent = (text, maxLength) => {
    if (text.length > maxLength) {
        return text.slice(0, maxLength - 3) + "..."
    }
    return text;
}

export const formatDate = (dateString, optionsObj) => {
    const defaultOptions = {
        year: "numeric",
        month: "short",
        day: "numeric",
        hour: "numeric",
        minute: "numeric"
    };

    const date = new Date(Date.parse(dateString));
    const formatter = new Intl.DateTimeFormat("en", optionsObj || defaultOptions);
    return formatter.format(date)
}