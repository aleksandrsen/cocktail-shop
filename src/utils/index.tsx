import React, { JSXElementConstructor, useEffect } from "react";
import { useLocation } from "react-router-dom";

export const ScrollTop = (): null => {
  const { pathname } = useLocation();

  useEffect(() => {
    window.scrollTo(0, 0);
  }, [pathname]);
  return null;
};

export const cutTextContent = (text: string, maxLength: number): string => {
  if (text.length > maxLength) return text.slice(0, maxLength - 3) + "...";
  return text;
};

export function getSkeletons(
  length: number,
  SkeletonComponent: JSXElementConstructor<any>,
  props: object = {}
): JSX.Element[] {
  return new Array(length)
    .fill(1)
    .map((num, idx) => <SkeletonComponent key={`${num}${idx}`} {...props} />);
}

type DateOptionItem = "2-digit" | "numeric";

type DateOptionsObj = {
  year?: DateOptionItem;
  month?: DateOptionItem | "narrow" | "short" | "long";
  weekday?: "narrow" | "short" | "long";
  day?: DateOptionItem;
  hour?: DateOptionItem;
  minute?: DateOptionItem;
};

export const formatDate = (
  date: string | Date | undefined,
  optionsObj?: DateOptionsObj
): string => {
  if (!date) return "";
  const defaultOptions = {
    year: "numeric",
    month: "short",
    day: "numeric",
    hour: "numeric",
    minute: "numeric",
  };

  const date_ = typeof date === "string" ? new Date(Date.parse(date)) : date;
  const formatter = new Intl.DateTimeFormat("en", optionsObj || defaultOptions);
  return formatter.format(date_);
};

export function searchByFields<T extends object, U extends keyof T>(
  arrayOfObj: T[],
  searchValue: string,
  keyFields: U[]
): T[] {
  const result: T[] = [];
  const term = searchValue.replace(",", " ").toLowerCase();

  arrayOfObj.forEach((obj: T) => {
    const descr = keyFields
      .reduce((arr: T[U][], field) => {
        arr.push(obj[field]);
        return arr;
      }, [])
      .join(" ");

    if (descr.toLowerCase().includes(term)) result.push(obj);
  });
  return result;
}

export const validateEmail = (str: string): boolean =>
  !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(str);

export const getTotalValue = <T extends object, K extends keyof T>(
  arr: T[],
  field: K
): number => {
  type Val = {
    [T[K]]: number
  }
  return arr.reduce((sum, item) => sum + item[field], 0)
};
