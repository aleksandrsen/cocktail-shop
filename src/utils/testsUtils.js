import { applyMiddleware, createStore } from "redux";
import { sagaMiddleware } from "../store";
import rootReducer from "../reducers";

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(sagaMiddleware)(
    createStore
  );
  return createStoreWithMiddleware(rootReducer, initialState);
};

export const findByTestAttr = (component, attr) =>
  component.find(`[data-test='${attr}']`);
