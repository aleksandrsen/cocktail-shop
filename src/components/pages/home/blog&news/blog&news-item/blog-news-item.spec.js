import React from "react";
import { shallow, render } from "enzyme";
import BlogNewsItem from "./blog-news-item";
import { cutTextContent, formatDate } from "../../../../../utils";

const setUp = (props = {}) => shallow(<BlogNewsItem post={props} />);

describe("BlogNewsItem component", () => {
  describe("Should render component without props", () => {
    let component;
    beforeEach(() => {
      component = setUp();
    });

    it("Should render BlogNewsItem component without errors", () => {
      const wrapper = component.find(".blogNewsItem");
      expect(wrapper.length).toBe(1);
    });

    it("Have not props", () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe("Should render component with props", () => {
    const props = {
      authorFullName: "Author",
      date: "2019-11-25T07:00",
      dislikes: 1,
      id: 2,
      likes: 3,
      title: "title",
      content: "description",
    };

    it("Have props", () => {
      const component = setUp(props);
      expect(component).toMatchSnapshot();
    });

    it("Should render correct long day number", () => {
      const component = setUp(props);
      const dayNum = component.find(".blogNewsItem__dateDay");
      expect(dayNum.text()).toBe(formatDate(props.date, { day: "numeric" }));
    });

    it("Should render correct short day number", () => {
      const props_ = { ...props, date: "2019-11-07T07:00" };
      const component = shallow(<BlogNewsItem post={props_} />);
      const dayNum = component.find(".blogNewsItem__dateDay");
      expect(dayNum.text()).toBe(
        0 + formatDate(props_.date, { day: "numeric" })
      );
    });

    it("Should render full text content", () => {
      const props_ = { ...props, content: "Short text content" };
      const component = setUp(props_);
      const textContent = component.find(".default-text");
      expect(textContent.text()).toBe(props_.content);
    });

    it("Should render cut text content", () => {
      const props_ = {
        ...props,
        content:
          "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci culpa explicabo maiores placeat rerum. Ad asperiores eos esse, harum, inventore iusto laborum necessitatibus nesciunt porro quae quas repellat tenetur, voluptate!",
      };
      const component = setUp(props_);
      const textContent = component.find(".default-text");
      expect(textContent.text()).toBe(cutTextContent(props_.content, 140));
    });
  });
});

//
// import React from "react";
// import { shallow, render } from "enzyme";
// import BlogNewsItem from "./blog-news-item";
// import {cutTextContent, formatDate} from "../../../../../utils";
//
// const setUp = (props = {}) => render(<BlogNewsItem post={props} />);
//
// describe("BlogNewsItem component", () => {
//   it("should render BlogNewsItem component without errors", () => {
//     const component = shallow(<BlogNewsItem />);
//     const wrapper = component.find(".blogNewsItem");
//     expect(wrapper.length).toBe(1);
//   });
//
//   describe("Have not props", () => {
//     let component;
//     beforeEach(() => {
//       component = setUp();
//     });
//     it("Should render 5 field skeletons", () => {
//       const skeletons = component.find(".fieldSkeleton");
//       expect(skeletons.length).toBe(4);
//     });
//
//     it("Should render img skeleton", () => {
//       const imgSkeleton = component.find(".imgSkeleton");
//       expect(imgSkeleton.length).toBe(1);
//     });
//   });
//
//   describe("Have props", () => {
//     let component;
//     const props = {
//       authorFullName: "Author",
//       date: "2019-11-25T07:00",
//       dislikes: 1,
//       id: 2,
//       likes: 3,
//       title: "title",
//       content: "description",
//     };
//
//     beforeEach(() => {
//       component = setUp(props);
//     });
//
//     it("Should render author full name", () => {
//       const authorFullName = component.find(".blogNewsItem__author span");
//       expect(authorFullName.text()).toBe(props.authorFullName);
//     });
//
//     it("Should render formatted date", () => {
//       const date = component.find(".blogNewsItem__dateMonth");
//       expect(date.text()).toBe(formatDate(props.date, { month: "short" }));
//     });
//
//     it("Should render correct long day number", () => {
//       const dayNum = component.find(".blogNewsItem__dateDay");
//       expect(dayNum.text()).toBe(formatDate(props.date, { day: "numeric" }));
//     });
//
//     it("Should render correct short day number", () => {
//       const props_ = { ...props, date: "2019-11-07T07:00" };
//       const component = shallow(<BlogNewsItem post={props_} />);
//       const dayNum = component.find(".blogNewsItem__dateDay");
//       expect(dayNum.text()).toBe(
//           0 + formatDate(props_.date, { day: "numeric" })
//       );
//     });
//
//     it("Should render full text content", () => {
//       const props_ = { ...props, content: "Short text content" };
//       const component = shallow(<BlogNewsItem post={props_} />);
//       const textContent = component.find(".default-text");
//       expect(textContent.text()).toBe(props_.content);
//     });
//
//     it("Should render cut text content", () => {
//       const props_ = {
//         ...props,
//         content: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci culpa explicabo maiores placeat rerum. Ad asperiores eos esse, harum, inventore iusto laborum necessitatibus nesciunt porro quae quas repellat tenetur, voluptate!",
//       };
//       const component = shallow(<BlogNewsItem post={props_} />);
//       const textContent = component.find(".default-text");
//       expect(textContent.text()).toBe(cutTextContent(props_.content, 140));
//     });
//   });
// });
