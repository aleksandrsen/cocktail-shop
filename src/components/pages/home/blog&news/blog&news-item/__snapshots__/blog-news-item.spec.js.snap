// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogNewsItem component Have not props 1`] = `
<div
  className="col col-12 blogNewsItem"
>
  <div
    className="row center"
  >
    <div
      className="col col-5 col-lg-6 col-md-12"
    >
      <div
        className="blogNewsItem__info"
      >
        <div
          className="blogNewsItem__date"
        >
          <div
            className="blogNewsItem__dateDay"
          >
            <FieldSkeleton
              styles={
                Object {
                  "height": "180px",
                  "width": "120px",
                }
              }
            />
          </div>
          <div
            className="blogNewsItem__dateMonth"
          />
        </div>
        <div
          className="blogNewsItem__details"
        >
          <FieldSkeleton
            styles={
              Object {
                "marginBottom": "16px",
                "padding": "16px 0",
                "width": "50%",
              }
            }
          />
          <FieldSkeleton
            styles={
              Object {
                "height": "90px",
                "marginBottom": "10px",
                "width": "100%",
              }
            }
          />
          <FieldSkeleton
            styles={
              Object {
                "padding": "15px",
                "width": "30%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="col col-5  col-lg-6 col-md-12"
    >
      <Memo(ImgSkeleton)
        classes={
          Array [
            "blogNewsItem__preview",
          ]
        }
        skeletonStyle={
          Object {
            "height": "250px",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`BlogNewsItem component Have props Have props 1`] = `
<div
  className="col col-12 blogNewsItem"
>
  <div
    className="row center"
  >
    <div
      className="col col-5 col-lg-6 col-md-12"
    >
      <div
        className="blogNewsItem__info"
      >
        <div
          className="blogNewsItem__date"
        >
          <div
            className="blogNewsItem__dateDay"
          >
            25
          </div>
          <div
            className="blogNewsItem__dateMonth"
          >
            Nov
          </div>
        </div>
        <div
          className="blogNewsItem__details"
        >
          <div
            className="blogNewsItem__author"
          >
            <svg
              height="16"
              width="16"
            >
              <use
                xlinkHref="#user-icon-home"
              />
            </svg>
            <span
              className="textOverflow"
            >
              Author
            </span>
          </div>
          <p
            className="default-text"
          >
            description
          </p>
          <div
            className="blogNewsItem__fullDate"
          >
            <svg
              height="16"
              width="16"
            >
              <use
                xlinkHref="#blog-news-clock"
              />
            </svg>
            <span>
              Nov 25, 2019
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col col-5  col-lg-6 col-md-12"
    >
      <Memo(ImgSkeleton)
        classes={
          Array [
            "blogNewsItem__preview",
          ]
        }
        skeletonStyle={
          Object {
            "height": "250px",
          }
        }
        title="title"
      />
    </div>
  </div>
</div>
`;

exports[`BlogNewsItem component Should render component with props Have props 1`] = `
<div
  className="col col-12 blogNewsItem"
>
  <div
    className="row center"
  >
    <div
      className="col col-5 col-lg-6 col-md-12"
    >
      <div
        className="blogNewsItem__info"
      >
        <div
          className="blogNewsItem__date"
        >
          <div
            className="blogNewsItem__dateDay"
          >
            25
          </div>
          <div
            className="blogNewsItem__dateMonth"
          >
            Nov
          </div>
        </div>
        <div
          className="blogNewsItem__details"
        >
          <div
            className="blogNewsItem__author"
          >
            <svg
              height="16"
              width="16"
            >
              <use
                xlinkHref="#user-icon-home"
              />
            </svg>
            <span
              className="textOverflow"
            >
              Author
            </span>
          </div>
          <p
            className="default-text"
          >
            description
          </p>
          <div
            className="blogNewsItem__fullDate"
          >
            <svg
              height="16"
              width="16"
            >
              <use
                xlinkHref="#blog-news-clock"
              />
            </svg>
            <span>
              Nov 25, 2019
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col col-5  col-lg-6 col-md-12"
    >
      <Memo(ImgSkeleton)
        classes={
          Array [
            "blogNewsItem__preview",
          ]
        }
        skeletonStyle={
          Object {
            "height": "250px",
          }
        }
        title="title"
      />
    </div>
  </div>
</div>
`;

exports[`BlogNewsItem component Should render component without props Have not props 1`] = `
<div
  className="col col-12 blogNewsItem"
>
  <div
    className="row center"
  >
    <div
      className="col col-5 col-lg-6 col-md-12"
    >
      <div
        className="blogNewsItem__info"
      >
        <div
          className="blogNewsItem__date"
        >
          <div
            className="blogNewsItem__dateDay"
          >
            <FieldSkeleton
              styles={
                Object {
                  "height": "180px",
                  "width": "120px",
                }
              }
            />
          </div>
          <div
            className="blogNewsItem__dateMonth"
          />
        </div>
        <div
          className="blogNewsItem__details"
        >
          <FieldSkeleton
            styles={
              Object {
                "marginBottom": "16px",
                "padding": "16px 0",
                "width": "50%",
              }
            }
          />
          <FieldSkeleton
            styles={
              Object {
                "height": "90px",
                "marginBottom": "10px",
                "width": "100%",
              }
            }
          />
          <FieldSkeleton
            styles={
              Object {
                "padding": "15px",
                "width": "30%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="col col-5  col-lg-6 col-md-12"
    >
      <Memo(ImgSkeleton)
        classes={
          Array [
            "blogNewsItem__preview",
          ]
        }
        skeletonStyle={
          Object {
            "height": "250px",
          }
        }
      />
    </div>
  </div>
</div>
`;
